pkg/oras/push.go

```diff
@@ -83,24 +83,29 @@ func pack(provider content.Provider, descriptors []ocispec.Descriptor, opts *pus
 	}
 
 	// Manifest
-	manifest := ocispec.Manifest{
-		Versioned: specs.Versioned{
-			SchemaVersion: 2, // historical value. does not pertain to OCI or docker version
-		},
-		Config:      config,
-		Layers:      descriptors,
-		Annotations: opts.manifestAnnotations,
-	}
-	manifestBytes, err := json.Marshal(manifest)
-	if err != nil {
-		return ocispec.Descriptor{}, nil, err
-	}
-	manifestDescriptor := ocispec.Descriptor{
-		MediaType: ocispec.MediaTypeImageManifest,
-		Digest:    digest.FromBytes(manifestBytes),
-		Size:      int64(len(manifestBytes)),
+	var manifestDescriptor ocispec.Descriptor
+	if opts.manifest == nil {
```

Comment:
if you invert this logic, you can improve readability:

```go
if opts.manifest != nil {
  return *opts.manifest, store, nil
}

manifest := ocispec.Manifest{
   ...
}
...
```

Reply:
I copied the syntax from the section above that allows overriding the `config` as well. I can update both?